### ISSUE TITLE
[DOCS] Unable to test compression

### ISSUE THREAD

**Issue by WolfspiritM** [minio/minio#10849]
We tried to get compression to work as the docs say.

https://docs.min.io/docs/minio-compression-guide.html
`To test this setup, practice put calls to the server using mc and use mc ls on the data directory to view the size of the object.`

But no matter what config we tried it always showed the actual size so we guessted compression wasn't working.
The problem is also that "mc admin config get cluster compression" shows off even so the environment variable turned it on.

I then tried to use a single node cluster and upload a high compressible file and it still stated it's 600MB but looking at the minio storage directory it was only around 40KB so compression does seem to work.

The docs are however misleading in this case.
Is there any other way to test if compression works for a cluster?

## Expected Behavior
"mc ls" shows the compressed size (as per docs) not the actual size

## Current Behavior
mc ls always shows the actual size (which is better but misleading in this test case)

## Possible Solution
Provide a way to also see the compressed size and/or update docs to remove the "To test the setup".
Also it would be nice if "mc admin config get local-cluster compression" shows the environment variables aswell.

## Steps to Reproduce (for bugs)
1. Upload a file with compression enabled
2. Check file size with mc ls

## Context
Enabling compression and checking if it is working

## Regression
No

## Your Environment

* Version used (`minio --version`): RELEASE.2020-11-06T23-17-07Z
* Server setup and configuration: Docker

**Comment by harshavardhana** [comment]
Listing always shows the actual size of the object this is to ensure AWS S3 compatibility

The object is compressed at rest, we don't add any other extra information here that leaks information. It is potentially possible to add an extra header to provide this information.  But nothing exists at this moment to know it's a compressed object.

**Comment by WolfspiritM** [comment]
@harshavardhana Then the "To test the setup" in the docs is very misleading especially the "mc ls" part. After reading it again I assume that it means to make a filesystem check in the actual local directory? I think that should be changed in the docs to make it more clear.

**Comment by harshavardhana** [comment]
> @harshavardhana Then the "To test the setup" in the docs is very misleading especially the "mc ls" part. After reading it again I assume that it means to make a filesystem check in the actual local directory? I think that should be changed in the docs to make it more clear.

Correct that's what it means 

> use mc ls on the data directory to view the size of the object.

We can add an extra metadata header like encryption to indicate it's a compressed object in future.

**Comment by WolfspiritM** [comment]
Sounds like a good idea. I think it would also be great when MINIO_COMPRESS is set to on that the mc admin config get command will indicate that it actually is on cause that made us guess that the env variable isn't working. Right now when using the env variable there is no indication that it's enabled at all

**Comment by harshavardhana** [comment]
> Sounds like a good idea. I think it would also be great when MINIO_COMPRESS is set to on that the mc admin config get command will indicate that it actually is on cause that made us guess that the env variable isn't working. Right now when using the env variable there is no indication that it's enabled at all

Env values are not present with mc admin config output, as mc admin config is meant for on disk values. Env values are not persisted on disk since env are self persistent.

**Comment by rherrmann** [comment]
I also ran into this issue. The docs still give a wrong advice:
> practice put calls to the server using mc and use mc ls on the data directory to view the size of the object.

It took me a while to find out that compression is actually working, but `mc ls` just shows the uncompressed size.

The documentation doesn't mention that compression is completely transparent to the outside either. Thus I made a few other attempts like `mc du`, `statObject` to get the  (compressed) size of an object.
